### DISCUSSION TITLE
Show build errors on the browser

### DISCUSSION THREAD

**Discussion by sergiodxa** [remix-run/remix#4196] [Proposals]
Right now, if you have a build error, and you reload the app, it will keep showing the old build instead of showing the build error on the browser, so you may think your changes didn't worked while in reality you're not receiving your changes at all.

The only way to find out your app has errors is by checking the terminal where you will see an error  on the build and then you can fix it.

I think Remix could catch those errors and send it to the browser so it shows an error screen like Create React App one ideally.

**Comment by kiliman** [comment]
I modified Remix to save build errors and RAS will display it on reload. It also supports `LiveReload`, so when you fix the error, it will auto-refresh.

Here's a video:

https://www.loom.com/share/39c6bd8ad3a94e2bbb9888b158f35c27

**Reply by jgentes** [reply]
Is this something that could be added as a PR, perhaps with a config option to enable it?

I've been caught quite a few times where I didn't realize the app wasn't reloading due to a build failure.

**Comment by harbar20** [comment]
Bump. I noticed this same issue, as outlined in #5174 . This is a severe missing feature of Remix; developers HAVE to know when and why their build fails in development.

**Reply by kiliman** [reply]
I've created a project with a patch to add this to Remix.

https://github.com/kiliman/remix-build-error

**Comment by joshpachner** [comment]
I would love it if Remix could add this. Maybe the devs creating remix don't write errors, but for us normal developers this is a must.

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
